game-app/game-core/src/main/java/games/strategy/triplea/ui/panel/move/MovePanel.java

```diff
@@ -158,63 +158,45 @@ public void unitsSelected(
 
         private void selectUnitsToMove(
             final List<Unit> units, final Territory t, final MouseDetails mouseDetails) {
-          // are any of the units ours, note - if no units selected that's still ok
-          if (!BaseEditDelegate.getEditMode(getData().getProperties())
-              || !selectedUnits.isEmpty()) {
-            for (final Unit unit : units) {
-              if (!unit.getOwner().equals(getUnitOwner(selectedUnits))) {
-                return;
-              }
-            }
+          final boolean editMode = BaseEditDelegate.getEditMode(getData().getProperties());
+
+          GamePlayer requiredOwner = null;
+          if (!editMode) {
+            requiredOwner = getCurrentPlayer();
+          } else if (!selectedUnits.isEmpty()) {
+            requiredOwner = selectedUnits.iterator().next().getOwner();
           }
```

Comment:
I split out this logic to make it much clearer what's being validated.

Reply:
If you add an else branch that assigns `null` to `requiredOwner` it can be made final.

Even better in my opinion would be to extract these couple of lines into a small helper method that returns an optional instance of `GamePlayer` (either wrapped in an optional, or annotated as `@Nullable`).

Reply:
Made a helper called `canSelectUnits()` instead, since that seemed cleaner.